wandb/sdk/wandb_run.py

```diff
@@ -3575,9 +3575,14 @@ def __getattr__(self, item: str) -> Any:
         self._assert_instance()
         return getattr(self._instance, item)
 
-    def wait(self) -> ArtifactInterface:
+    def wait(self, wait_timeout_secs: int = None) -> ArtifactInterface:
         if not self._instance:
-            resp = self._future.get().response.log_artifact_response
+            future_get = self._future.get(wait_timeout_secs)
+            if not future_get:
+                raise TimeoutError(
```

Comment:
@speezepearson is the desired outcome a raised error?

Reply:
I think that's the only way to make this change backwards-compatible! Traditionally (like, in the [`threading`](https://docs.python.org/3/library/threading.html) module), a timeout results in returning `False` instead of `True`; but we can't change this return type. (Or, well, I guess we could return an `Optional[ArtifactInterface]`, and only return None if it times out? I wouldn't object to that, but I don't see any advantages it has over the raise-an-exception solution.)

Reply:
Either is fine by me!